podman_compose.py

```diff
@@ -987,25 +1016,42 @@ def flat_deps(services, with_extends=False):
     create dependencies "_deps" or update it recursively for all services
     """
     for name, srv in services.items():
-        deps = set()
-        srv["_deps"] = deps
+        deps = {}
         if with_extends:
             ext = srv.get("extends", {}).get("service", None)
             if ext:
                 if ext != name:
-                    deps.add(ext)
+                    deps[ext] = DependsCondition.STARTED
                 continue
-        deps_ls = srv.get("depends_on", None) or []
+        deps_ls = srv.get("depends_on", [])
         if is_str(deps_ls):
-            deps_ls = [deps_ls]
+            # depends_on: "foo"
+            # treat as condition: service_started
+            deps_ls = {deps_ls: DependsCondition.STARTED}
         elif is_dict(deps_ls):
-            deps_ls = list(deps_ls.keys())
-        deps.update(deps_ls)
+            # depends_on:
+            #   foo:
+            #       condition: service_xxx
+            tmp = {}
+            for service, condition in deps_ls.items():
+                condition = DependsCondition.to_enum(condition.get("condition"))
+                tmp[service] = condition
+            deps_ls = tmp
+        else:
+            # depends_on:
+            # - foo
+            # treat as condition: service_started
+            deps_ls = {dep: DependsCondition.STARTED for dep in deps_ls}
+        deps = {**deps, **deps_ls}
         # parse link to get service name and remove alias
-        links_ls = srv.get("links", None) or []
+        links_ls = srv.get("links", [])
```

Comment:
The style or `mydict.get(key, None) or []` was intentional to handle cases like being empty string qnd i want it to be empty array (not an array of single element that is empty string)

Reply:
ok, I will revert these changes then and maybe add a comment for future reference